runtime/drivers/athena/athena.go

```diff
@@ -0,0 +1,314 @@
+	m := make(map[string]any, 0)
+	_ = mapstructure.Decode(c.config, m)
+	return m
+}
+
+// Close implements drivers.Connection.
+func (c *Connection) Close() error {
+	return nil
+}
+
+// Registry implements drivers.Connection.
+func (c *Connection) AsRegistry() (drivers.RegistryStore, bool) {
+	return nil, false
+}
+
+// Catalog implements drivers.Connection.
+func (c *Connection) AsCatalogStore(instanceID string) (drivers.CatalogStore, bool) {
+	return nil, false
+}
+
+// Repo implements drivers.Connection.
+func (c *Connection) AsRepoStore(instanceID string) (drivers.RepoStore, bool) {
+	return nil, false
+}
+
+// OLAP implements drivers.Connection.
+func (c *Connection) AsOLAP(instanceID string) (drivers.OLAPStore, bool) {
+	return nil, false
+}
+
+// Migrate implements drivers.Connection.
+func (c *Connection) Migrate(ctx context.Context) (err error) {
+	return nil
+}
+
+// MigrationStatus implements drivers.Connection.
+func (c *Connection) MigrationStatus(ctx context.Context) (current, desired int, err error) {
+	return 0, 0, nil
+}
+
+// AsObjectStore implements drivers.Connection.
+func (c *Connection) AsObjectStore() (drivers.ObjectStore, bool) {
+	return c, true
+}
+
+// AsTransporter implements drivers.Connection.
+func (c *Connection) AsTransporter(from, to drivers.Handle) (drivers.Transporter, bool) {
+	return nil, false
+}
+
+func (c *Connection) AsFileStore() (drivers.FileStore, bool) {
+	return nil, false
+}
+
+// AsSQLStore implements drivers.Connection.
+func (c *Connection) AsSQLStore() (drivers.SQLStore, bool) {
+	return nil, false
+}
+
+func cleanPath(ctx context.Context, cfg aws.Config, bucketName, prefix string) error {
+	s3client := s3v2.NewFromConfig(cfg)
+	out, err := s3client.ListObjectsV2(ctx, &s3v2.ListObjectsV2Input{
+		Bucket: &bucketName,
+		Prefix: &prefix,
+	})
+	if err != nil {
+		return err
+	}
+
+	ids := make([]s3v2types.ObjectIdentifier, 0, len(out.Contents))
+	for _, o := range out.Contents {
+		ids = append(ids, s3v2types.ObjectIdentifier{
+			Key: o.Key,
+		})
+	}
+	_, err = s3client.DeleteObjects(ctx, &s3v2.DeleteObjectsInput{
+		Delete: &s3v2types.Delete{
+			Objects: ids,
+		},
+	})
+	return err
+}
+
+func (c *Connection) DownloadFiles(ctx context.Context, source *drivers.BucketSource) (drivers.FileIterator, error) {
+	conf, err := parseSourceProperties(source.Properties)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	cfg, err := awsconfig.LoadDefaultConfig(
+		ctx,
+		awsconfig.WithRegion(conf.Region),
+		awsconfig.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(c.config.AccessKeyID, c.config.SecretAccessKey, c.config.SessionToken)),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	prefix := "parquet_output_" + uuid.New().String()
+	bucketName := strings.TrimPrefix(strings.TrimRight(conf.OutputLocation, "/"), "s3://")
+	unloadPath := bucketName + "/" + prefix
+	err = c.unload(ctx, cfg, conf, "s3://"+unloadPath)
```

Comment:
See earlier note – it should take `OutputLocation` in the `s3://bucket/path` format, and then we can use `url.Parse` to parse it and obtain the bucket name for `openBucket`.

We should also make sure it supports nested output locations, as far as I can tell, it can't be assumed the bucket is dedicated to only Athena?